### PR TITLE
Sion backend v2

### PR DESCRIPTION
**Pull request by wschenck**

### CODE REVIEW

File: examples/test-reader.py [hunk 1]
```diff
@@ -43,11 +51,13 @@ def write(fstr, *vals):
 #>> row={}
 #""", np.array_repr(j))
         gid, step, offset = j[0], j[1], j[2]
-        v = j[3] if i.values else None
+        double_v = j[3] if i.double_n_val else None
```
Comment (apeyser) [inline examples/test-reader.py]: Maybe these two lines would be clearer in an if-else block:
```
if i.double_n_val:
    # set double_v
    if i.long_n_val:
        # set long_v
     else:
        # set long_v
else:
    # set long_v

**Review by apeyser — COMMENTED** [review]
Only one if-else question, otherwise looks good (and was discussed earlier)

File: examples/test-reader.py (hunk 1 above)
Comment (wschenck) [reply examples/test-reader.py]: Ok, but one also has to set double_v to None in the outer else block. (like it has to be done for long_v in the inner else block)

Btw., the outer else block has to contain an if-else for long_v as well.

File: examples/test-reader.py [hunk 2]
```diff
@@ -51,8 +51,13 @@ def write(fstr, *vals):
 #>> row={}
 #""", np.array_repr(j))
         gid, step, offset = j[0], j[1], j[2]
-        double_v = j[3] if i.double_n_val else None
-        long_v = j[4] if i.long_n_val and i.double_n_val else j[3] if i.long_n_val else None
+
+        double_v, long_v = {(False, False): lambda: (None, None),
```
Comment (apeyser) [inline examples/test-reader.py]: How about this?

Comment (apeyser) [reply examples/test-reader.py]: To test this, which sha on NEST should I use to produce the *.sion?

Comment (wschenck) [reply examples/test-reader.py]: Your solution works, I have just tested it with an existing sionlib output file in which all cases are represented.

Comment (wschenck) [reply examples/test-reader.py]: Btw, the SHA is 7aa979e8d0e2d30.
But then you have to write a simulation script with the spin_detector to cover all cases, and for the spin_detector you need special neurons to connect to it.

**Review by apeyser — APPROVED** [review]

**Pull request merged**